tests/ui/macros/issue-111749.rs

```diff
@@ -5,8 +5,8 @@ macro_rules! cbor_map {
 }
 
 fn main() {
-    cbor_map! { #[test(test)] 4};
-    //~^ ERROR removing an expression is not supported in this position
+    cbor_map! { #[test(test)] 4i32};
+    //~^ ERROR the `#[test]` attribute may only be used on a non-associated function
```

Comment:
I think this is the correct error here, not something about removing expressions. I think noting that the target is wrong is more important.